cloudant-client/src/main/java/com/cloudant/client/api/DesignDocumentManager.java

```diff
@@ -151,8 +156,21 @@ public DesignDocument get(String id, String rev) {
      */
     public Response remove(String id) {
         assertNotEmpty(id, "id");
-        DesignDocument find = db.find(DesignDocument.class, id);
-        return db.remove(find.getId(), find.getRevision());
+        String revision = null;
+        try {
+            // Get the revision ID from ETag, removing leading and trailing "
+            revision = client.executeRequest(Http.HEAD(new DatabaseURIHelper(db.getDBUri()
+            ).documentUri(id))).execute().getConnection().getHeaderField("ETag");
+        } catch (IOException ioe) {
+            throw new RuntimeException("There was an error getting revision information so the " +
+                    "design document cannot be removed.", ioe);
+        }
+        if (revision != null) {
+            revision = revision.substring(1, revision.length() - 1);
+            return db.remove(id, revision);
+        } else {
+            throw new RuntimeException("No ETag header found for design document with id " + id);
+        }

```

Comment:
Added in cd6adb5. It helpfully demonstrated that I need not catch an `IOException` because I didn't need to call `execute` as `executeRequest` was already doing that.
Also changed the no ETag case to use a `CouchDbException` rather than a generic `RuntimeException` because if there is no ETag header that really is a server problem.